flashlight/autograd/Functions.h

```diff
@@ -361,6 +363,18 @@ split(const Variable& input, const std::vector<dim_t>& splitSizes, int dim);
  */
 Variable tile(const Variable& input, const af::dim4& dims);
 
+/**
+ * Repeats the tensor `input` along specific dimensions. The number of
+ * repetition along each dimension is specified in descriptor `dims`.
+ *
+ * @param[in] precision Type of the output vector when is it is desired to
+ * be different from the input type. This is particularly useful when tile is
+ * applied on parameters and the results will be used in a half precision
+ * arithmetic.
+ */
+Variable
+tile(const Variable& input, const af::dim4& dims, const af::dtype precision);
```

Comment:
Yes, sometimes we use this function to tile the weights. In such settings, it would be beneficial to use the target type instead of weight type (i.e., f32). Please see line: 41 in  LayerNorm.cpp.
The function is overloaded. The default one (line 364 in Functions.h) inspects the input type and performs the tilling according to that.